Add vizro plotly templates

huong-li-nguyen opened the pull request:
Hello @AnnMarieW  👋 

- Adds `vizro_light.json` and `vizro_dark.json`: These files are sourced directly from [here](https://github.com/mckinsey/vizro/tree/main/vizro-core/src/vizro/_themes). They are currently formatted according to our configurations, but I can reformat them to match your repository's style if you prefer the original formatting. Just let me know 👍
- Updated certain specifications in `dbc.css`: The `border-radius` setting ensures that components align with the overall Bootstrap theme design. For instance, in themes without rounded borders (such as vizro), the components will also lack rounded borders. I verified that this change does not affect other Bootstrap themes unless they diverge from the default. I also adjusted the slider tooltip colors to align with the primary/secondary color palette, resulting in a more consistent appearance overall. 

**Context**
This is a PR in preparation for the upcoming release of vizro-bootstrap. Once [this PR](https://github.com/mckinsey/vizro/pull/970) is merged, users will be able to import the `vizro-bootstrap` theme using the command below. I've also updated our documentation in that PR, referencing this repository for guidelines on how to apply the Bootstrap theme to Plotly charts and other Dash components :) 

Feel free to take a look as well!

```
import vizro
from dash import Dash

app = Dash(__name__, external_stylesheets=[vizro.bootstrap])
```


**TO DO**

- [x] add a minimal example app in the /examples folder,
- [x] update the init to include vizro templates if ALL is selected.
- [x] Include some notes in _create_templates.py to say that vizro templates need to be updated manually.
- [x] changelog
- [x] update the Theme Explorer site for the new Vizro templates after release @AnnMarieW

huong-li-nguyen committed:
Add vizro plotly templates

huong-li-nguyen committed:
Update dbc.css

- Apply default border radius to dropdown and slider tooltip
- Change track color to full opacity
- Change slider tooltip color to bs-secondary

huong-li-nguyen commented on dbc.css:
```diff
@@ -277,6 +277,10 @@ body .dbc .dash-table-container .dash-spreadsheet-container .dash-spreadsheet-in
+.dbc .rc-slider-tooltip-inner {
+  background-color: var(--bs-secondary);
+  border-radius: var(--bs-border-radius);
+}
```
I've checked for a few bootstrap themes. Using the bs-border-radius variable doesn't seem to affect them, the colors of the slider tooltip are now more consistent with the overall theme:

![Screenshot 2025-01-30 at 13 43 34](https://github.com/user-attachments/assets/48deb7e9-a1dc-4a58-8953-a76641568094)
![Screenshot 2025-01-30 at 13 43 27](https://github.com/user-attachments/assets/dcd6e5bb-0335-45a2-bf79-1c187a280e3f)
![Screenshot 2025-01-30 at 13 43 48](https://github.com/user-attachments/assets/16a7a47c-35fa-43bb-a975-aa1bcb3f2d97)

huong-li-nguyen commented on src/dash_bootstrap_templates/templates/vizro.json:
```diff
@@ -0,0 +1,272 @@
+{
```
Let me know if you don't like the formatting of this file, I can also change it back to the unformatted version 👍

huong-li-nguyen commented on dbc.css:
```diff
@@ -246,7 +246,7 @@ body .dbc .dash-table-container .dash-spreadsheet-container .dash-spreadsheet-in
 
 .dbc .rc-slider-track {
-  background-color: rgba(var(--bs-primary-rgb), 0.50);
+  background-color: var(--bs-primary-rgb);
```
Just a suggestion: For active traces/selections (such as the selected range track here), our designer normally uses the primary colour in full opacity. This would also make it more consistent with the selected color on the checkboxes. However, this might not be to everyone's taste. So, I'm happy to revert the change if you think it doesn't look better overall 👍

AnnMarieW commented:
Hi Li!  Thanks for the PR :slightly_smiling_face: 

The Vizro templates look fantastic!  I also like the updates to the dbc.css

I recently did another release -- V2.0 now requires Plotly 6.0.  This change is from another great community PR. 
 https://github.com/AnnMarieW/dash-bootstrap-templates/pull/46

Do you know if your vizro templates are compatible with Plotly 6?  Will Plotly 6 be an issue with Vizro apps in general?

I just saw the PR to publish the Vizro Bootstrap stylesheet :tada:   I would like to test that with the vizro templates  here before we merge this PR.  

There will be a few more things to update later:
 - add a minimal example app in the /examples folder,
 - update the init to include vizro templates if ALL is selected.  
 - Include some notes in `_create_templates.py` to say that vizro templates need to be updated manually.
 - changelog
 - (for me) update the Theme Explorer site for the new Vizro templates after release

huong-li-nguyen commented:
> Do you know if your vizro templates are compatible with Plotly 6? Will Plotly 6 be an issue with Vizro apps in general?
> 
> I just saw the PR to publish the Vizro Bootstrap stylesheet 🎉 I would like to test that with the vizro templates here before we merge this PR.
> 
> There will be a few more things to update later:
> 
> * add a minimal example app in the /examples folder,
> * update the init to include vizro templates if ALL is selected.
> * Include some notes in `_create_templates.py` to say that vizro templates need to be updated manually.
> * changelog
> * (for me) update the Theme Explorer site for the new Vizro templates after release

Hey @AnnMarieW ! 

Thanks for the quick update! :) Let's merge this PR first then: https://github.com/mckinsey/vizro/pull/970
And in the mean-time, I tackle the remaining TODOs here 👍 

The templates should be compatible with Plotly 6, as we do not use any of the deprecated attributes in the template. I just double-checked again 👍 I've also changed that our visual-vocabulary will run on plotly 6.0.0 and we also have some tests in our repository.

huong-li-nguyen committed:
PR comments

huong-li-nguyen commented on CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## Feb X, 2025
```
TODO: Update after final check

huong-li-nguyen commented on examples/demo_vizro_bootstrap.py:
```diff
@@ -0,0 +1,37 @@
+
+from dash_bootstrap_templates import load_figure_template
+
+app = Dash(__name__, external_stylesheets=[vizro.bootstrap])
```
TODO: Run script and check after release of vizro.bootstrap: https://github.com/mckinsey/vizro/pull/970

huong-li-nguyen replied:
Tested with packaged whl file after merge of above PR ✅

huong-li-nguyen committed:
Update chart templates

huong-li-nguyen committed:
Update vizro version to where vizro-bootstrap is public

huong-li-nguyen committed:
Improve example app

huong-li-nguyen committed:
Simplify color mode switch

huong-li-nguyen commented on examples/demo_vizro_bootstrap.py:
```diff
@@ -0,0 +1,77 @@
+"""
```
The example looks like this :)

![Screenshot 2025-02-04 at 15 23 48](https://github.com/user-attachments/assets/321334b0-c762-4e31-a788-14730027decc)

![Screenshot 2025-02-04 at 15 23 41](https://github.com/user-attachments/assets/53e658a1-e2f3-4a9d-923c-c68151211067)

AnnMarieW replied:
This looks great!

huong-li-nguyen commented on setup.cfg:
```diff
@@ -32,6 +32,7 @@ install_requires =
     importlib_resources>=5.1.0; python_version < "3.9"
     numpy
     plotly>=6,<7
+    vizro>=0.1.32
```
I wasn't sure whether I should add the dependency here or not 🤔 They need vizro installed for the example to work, but it's not really required for `dash_bootstrap_templates`.

AnnMarieW replied:
I think that will make it so that everyone who uses dash_bootstrap_templates will have to install vizro.  It would be best to just include that as a comment in the example code.

huong-li-nguyen commented:
> There will be a few more things to update later:
> 
> * add a minimal example app in the /examples folder,
> * update the init to include vizro templates if ALL is selected.
> * Include some notes in `_create_templates.py` to say that vizro templates need to be updated manually.
> * changelog
> * (for me) update the Theme Explorer site for the new Vizro templates after release


Hey @AnnMarieW,

I think everything is done from my side on this PR :) I just tested it with a local whl file, but you will be able to test the example properly after we have released `vizro==0.1.32`. 

EDIT: It just got released: https://github.com/mckinsey/vizro/releases/tag/0.1.32 🥳

AnnMarieW commented:
Thanks for the PR @huong-li-nguyen 

I had a quick look and I thinks it looks good. I just want to spend a little more time with it -- I'll get back to this a few days.  I'm working on a  pre-release for dash-mantine-components  that will be compatible with Dash 3.0.  Need to get that over the finish line first :-)

antonymilne commented:
Just a quick note to say that I've been watching this and it all looks great 😀 Thank you @huong-li-nguyen and @AnnMarieW for your work on this!

AnnMarieW commented:
Hi @huong-li-nguyen 

In the Vizro example app we should also show how to get the URL for the Vizro Bootstrap stylesheet if people are using the stylesheet in a dash app without using any other Vizro features.  


>install vizro
print(vizro.bootstrap)

```
vizro_bootstrap = "https://cdn.jsdelivr.net/gh/mckinsey/vizro@0.1.33/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css"
app = Dash(__name__, external_stylesheets=[vizro_bootstrap])
```

This will be necessary if/when dependencies get out of sync.  For example, you can't currently use both Vizro and Dash 3

Also, need to note that the Vizro Bootstrap is not compatible with the `ThemeSwitchAIO` or the `ThemeChangerAIO` but does work with the [Bootstrap color modes](https://hellodash.pythonanywhere.com/adding-themes/color-modes), which is the recommended method of switching between light and dark themes. (And is demonstrated in the app)


**UPDATE:**
I forgot that we made it possible to add custom stylesheets to the `ThemeChangerAIO`

It can be added like this:

```

theme_changer = ThemeChangerAIO(
    aio_id="theme",
    button_props={"color": "primary", "outline": True}, 
    custom_themes={'vizro': vizro_boostrap},
)

```

But it's still better to use Bootstrap color modes rather than the `ThemeSwitchAIO`  That theme switch component was developed before Bootstrap had the color modes feature and was typically used to switch between a light and dark Bootstrap themes.

AnnMarieW commented on CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## Feb X, 2025
+Include Vizro plotly templates after release of [vizro.bootstrap](https://github.com/mckinsey/vizro/pull/970)
+ - Add `vizro_dark.json` and `vizro.json` plotly templates
```
In the changelog, let's include a link to the vizro github https://github.com/mckinsey/vizro/tree/main/vizro-core/src/vizro/_themes and say which version was used.

AnnMarieW commented:
I'm working on adding the Vizro stylesheets and templates to the Dash Theme Explorer site.  Everything is looking great so far except for a colormode switch that I use in a lot of examples.   I've seen it used "in the wild", so it would be great if it looked good with the Vizro theme too.  

With all other Bootstrap themes, the icons are in-line with the switch.  Do you know how to fix it?

![image](https://github.com/user-attachments/assets/043d51bb-aa16-4288-9c85-249555612004)


```python
from dash import Dash, html
import dash_bootstrap_components as dbc

vizro_bootstrap = "https://cdn.jsdelivr.net/gh/mckinsey/vizro@0.1.33/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css"

app = Dash(__name__, external_stylesheets=[dbc.icons.FONT_AWESOME, vizro_bootstrap])

color_mode_switch =  html.Span(
    [
        dbc.Label(className="fa fa-moon", html_for="switch"),
        dbc.Switch( id="switch", value=False, className="d-inline-block ms-1"),
        dbc.Label(className="fa fa-sun", html_for="switch"),
    ]
)

app.layout = dbc.Container(
    [
        html.H3(["Bootstrap Light Dark Color Modes Demo"], className="bg-primary p-2"),
        color_mode_switch,
    ]
)

if __name__ == "__main__":
    app.run_server(debug=True)


```

huong-li-nguyen committed:
Update demo_vizro_bootstrap.py

huong-li-nguyen committed:
PR comments

huong-li-nguyen committed:
Add more comments

huong-li-nguyen commented:
Hey @AnnMarieW,


thanks for raising the issue! I just created a PR on the vizro repository to fix it. Afterwards, I'll initiate a release, so this should be then fixed in `vizro>=0.1.34` 👍 

**A few changes I've added:**
- Removed the vizro dependency form the `setup.cfg` and added it as a comment inside the demo vizro app
- Added your moon / sun example directly into the demo vizro app as well
- Added a comment on how to pull the css file directly if they do not want to import vizro inside the docstring of the vizro demo app
- Updated the changelog file with the vizro version required for both `vizro.bootstrap` and the Plotly templates

This should all work after merging https://github.com/mckinsey/vizro/pull/1012. You should then be able to test it out via:

```
vizro_bootstrap = "https://cdn.jsdelivr.net/gh/mckinsey/vizro@main/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css"
app = Dash(__name__, external_stylesheets=[vizro_bootstrap])
```

huong-li-nguyen committed:
Change to latest CSS file

huong-li-nguyen committed:
Pull latest chart templates

huong-li-nguyen commented on examples/demo_vizro_bootstrap.py:
```diff
@@ -0,0 +1,102 @@
+app = Dash(__name__, external_stylesheets=[vizro.bootstrap])
+
+# Alternatively, you could do:
+# vizro_bootstrap = "https://cdn.jsdelivr.net/gh/mckinsey/vizro@main/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css"
```
Seems like the cdn link is still outdated even after merge. But if you want to test it out already, then try this link: "https://cdn.jsdelivr.net/gh/mckinsey/vizro@bug/fix-bootstrap-toggle-switch/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css"

The link here should work after release though 👍

huong-li-nguyen committed:
Add dbc.icons

AnnMarieW commented:
Hi @huong-li-nguyen 

This looks great!  

One small tweak to the Changelog - it should show the exact version that was used when the templates were added to the file.  So this line should just be `=` rather than `>=`
```
corresponding to chart templates available in `vizro>=0.1.34`
```

I have a PR for the Theme Explorer site ready to go.  I'd like to get your feedback on it, but it might be easier to update it once it's live rather than work with that PR.  I also made some tweaks to make it work better with all themes including Vizro, like removing the `text-white` that was in some of the headings.

Let me know when this PR is ready to merge, then it would be great to do a forum post to introduce this new feature :slightly_smiling_face:

antonymilne commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+## Feb 12, 2025
+ - Integrated Vizro Plotly templates following the release of [vizro.bootstrap](https://github.com/mckinsey/vizro/pull/970).
```
```suggestion
 - Integrated Vizro Plotly templates following the release of [`vizro.bootstrap`](https://github.com/mckinsey/vizro/pull/970).
```

antonymilne commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+Using `vizro.bootstrap` requires `vizro>=0.1.34` to be installed.
+ - Added `vizro_dark.json` and `vizro.json` plotly templates from [vizro themes](https://github.com/mckinsey/vizro/tree/main/vizro-core/src/vizro/_themes),
+corresponding to chart templates available in `vizro>=0.1.34`.
+ - Applied minor visual adjustments to `dbc.css`
```
```suggestion
 - Applied minor visual adjustments to `dbc.css`.
```

antonymilne reviewed (commented):
I just had another look through this and it's still great 😀

AnnMarieW committed:
Update CHANGELOG.md

Co-authored-by: Antony Milne <49395058+antonymilne@users.noreply.github.com>

AnnMarieW committed:
Update CHANGELOG.md

Co-authored-by: Antony Milne <49395058+antonymilne@users.noreply.github.com>

AnnMarieW commented:
> I just had another look through this and it's still great 😀

I agree!  It's my new favorite theme on the Dash Bootstrap Theme Explorer site. 

Li and the whole Vizro team did an amazing job on the figure templates and Bootstrap theme.   I'm thrilled you made this available to the community.  I can't wait till it's live :partying_face:

huong-li-nguyen committed:
PR comment

huong-li-nguyen commented:
> Li and the whole Vizro team did an amazing job on the figure templates and Bootstrap theme. I'm thrilled you made this available to the community. I can't wait till it's live 🥳

Thanks for starting off the journey with us back then! ❤️ 

I just incorporated your last PR suggestion and tested it out with the latest release `vizro==0.1.34`. So we can merge this PR now and you can properly test with the latest vizro version 🥳

AnnMarieW commented:
Hi @huong-li-nguyen 
I just tried it with vizro==0.1.34 and the color mode switch still looks wrong,  I also tried it in  a different browser in case the css was cached and it still looked the same.  

**Update:**
It's working in a stand-alone app, just not in the Theme Explorer site.  I may have some conflicting CSS.  Taking a look now.

Ok, I found it.  Working now :tada:

huong-li-nguyen commented:
> Hi @huong-li-nguyen I just tried it with vizro==0.1.34 and the color mode switch still looks wrong, I also tried it in a different browser in case the css was cached and it still looked the same.
> 
> **Update:** It's working in a stand-alone app, just not in the Theme Explorer site. I may have some conflicting CSS. Taking a look now.
> 
> Ok, I found it. Working now 🎉

Amazing!!! 🚀

AnnMarieW commented:
@huong-li-nguyen @antonymilne 

The Theme Explorer site is live now  https://hellodash.pythonanywhere.com/adding-themes/vizro-bootstrap

Please consider this a draft and I'd be happy to update it based on your feedback.   That includes any of the sections - not just things related to Vizro :slightly_smiling_face: 

One thing I've notices with the Vizro theme is that the markdown text is a little compressed.  Does this look right to you or might I have some other conflicting CSS?

huong-li-nguyen commented:
> @huong-li-nguyen @antonymilne
> 
> The Theme Explorer site is live now https://hellodash.pythonanywhere.com/adding-themes/vizro-bootstrap
> 
> Please consider this a draft and I'd be happy to update it based on your feedback. That includes any of the sections - not just things related to Vizro 🙂
> 
> One thing I've notices with the Vizro theme is that the markdown text is a little compressed. Does this look right to you or might I have some other conflicting CSS?

Hey @AnnMarieW,

ah, this looks great to me! I also noticed the compressed text, it's coming from our BS theme. Let me actually fix that! 

Apart from that: 

- "Learn more at Vizro" -> Could we link to the vizro github repository here? :)
- Could we add a bit of gap between the two images on the top?

Other than that, it's good from my side, but i'll let @antonymilne take a look as well, while I fix the compressed text 😄

antonymilne commented:
Hi @AnnMarieW! This looks really great, thank you so much 🙏 If you'd like then I can make suggestions directly in a PR, but it seems like the code for the website itself isn't available on this repo?

In brief though, here's some small bits I'd consider changing:
1. "Learn more at [Vizro](https://hellodash.pythonanywhere.com/adding-themes/vizro-bootstrap), and the [Visual Vocabulary](https://vizro-demo-visual-vocabulary.hf.space/) site." ➡️ I'd remove this completely because the first link goes to the same page (?) and the second is just one particular dashboard. If you'd like to link to some Vizro examples then I'd instead do [Vizro examples gallery](http://vizro.mckinsey.com/).
2. Link to Vizro in "What is Vizro?" should go to https://github.com/mckinsey/vizro
3. "Vizro Features Available for Dash Apps" section, Vizro Bootstrap theme point - maybe worth putting a cross-link to our docs https://vizro.readthedocs.io/en/stable/pages/user-guides/themes/#vizro-bootstrap-in-a-pure-dash-app. And we'll put some reciprocal links into your pages and demo when it's ready. Just so everything joins up nicely and people can cross-reference bits easily.
4. Vizro Bootstrap Figure Templates section:
    1. `dash-bootstrap-templates>=V2.1.0` - remove the `V` or I think `pip install` won't like it.
    2.  FYI we actually have an alternative that's independent from your `load_figure_template` function. Doing `import vizro` registers the plotly themes `"vizro_light"` and `"vizro_dark"`. The end result for the user will be identical because the plotly theme json files we put here are identical to those built into vizro. But it might be worth mentioning this alternative method and/or linking to [the docs on it](https://vizro.readthedocs.io/en/stable/pages/user-guides/themes/#vizro-themes-in-plotly-charts) - up to you.
5. "The vizro.bootstrap is a is a predefined URL"  ➡️ repeated "is a"
6. "vizro_boostrap" typo with missing "t" in a couple of places (good to see I'm not the only one who constantly does this 😅)

And one bigger point where I'm not sure what to do: my personally "preferred" way to use vizro bootstrap would be to `import vizro` and `vizro.bootstrap`, since this version is bumped on every vizro release and so will give you the latest stable stylesheet with bugfixes etc. But it's also useful for people to have a static link I know.

Maybe what I'd suggest is we make the `vizro.bootstrap` the primary (uncommented/recommended) approach in all examples (both on https://hellodash.pythonanywhere.com/adding-themes/vizro-bootstrap and the "Sample App" page) and then mention commented after that `vizro.bootstrap` just contains a static URL and show how that can be used instead. This would be consistent with how [dbc Themes are documented](https://dash-bootstrap-components.opensource.faculty.ai/docs/themes/), which is basically the pattern we were following with vizro bootstrap.

If you'd prefer to keep the preference order the same way you have it now (static link first, then `vizro.bootstrap` afterwards) then that's fine too but I'm not sure what static URL you should use - currently you have it pinned to 0.1.34 which will go out of date. Maybe better to use `@main` which hopefully gets the latest stable version instead.

Edit: case in point here is that @huong-li-nguyen just fixed the line height, and it would be nice to propagate that change through to your demo without needing to re-release vizro.

huong-li-nguyen commented:
> If you'd prefer to keep the preference order the same way you have it now (static link first, then `vizro.bootstrap` afterwards) then that's fine too but I'm not sure what static URL you should use - currently you have it pinned to 0.1.34 which will go out of date. Maybe better to use `@main` which hopefully gets the latest stable version instead.
> 
> Edit: case in point here is that @huong-li-nguyen just fixed the line height, and it would be nice to propagate that change through to your demo without needing to re-release vizro.


I noticed that `https://cdn.jsdelivr.net/gh/mckinsey/vizro@main/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css` doesn't always update properly even after everything has been merged and I manually tried to purge.

What seems to work though is adding `?v=2` which seems to force fetch the latest version. So I would recommend using this link

`https://cdn.jsdelivr.net/gh/mckinsey/vizro@main/vizro-core/src/vizro/static/css/vizro-bootstrap.min.css?v=2`

instead of the pinned version one :) Then it should hopefully be always up-to-date 🤞

AnnMarieW commented:
Hi @antonymilne and @huong-li-nguyen 
Thank you so much for your detailed review!  You just reminded me to scan that entire repo for "boostrap".  Again.

I'll make all the copy edit changes, but I'm struggling with the right way to use the Vizro Bootstrap.  The main issue when using `import vizro` is the dependencies.  If you are using it in a dash app (rather than a vizro app) then you may need to run a different version of dash-ag-grid,  dmc, or dash etc.  That's possible if you just specify the link, rather than using `vizro.bootstrap`  

So for now I'll make it  clear that the preferred method is to use `vizro.bootstrap` but the examples in my site are live, so I'll have to use the link method and put the preferred method in comments.

antonymilne commented:
@AnnMarieW your concern about dependency conflict is definitely well-founded 👍 All I can say is that we like to keep vizro quite open ended with its dependencies (we only upper cap in exceptional cases, like the dmc one) to ensure maximum compatibility with other dash libraries. But I can indeed imagine there might be some cases where someone wants to use an old version of `dmc` for example that's not compatible with vizro and hence can't get the vizro bootstrap theme through `import vizro`. In this case their best bet would indeed be to the use the static URL like you say.

In reality I think the best solution, as ever, is to pin your dependencies in production. My suggested way to do this would be to set `vizro>=0.1.34` in the `requirements.in` file, use `vizro.bootstrap`, and freeze requirements to a locked `requirements.txt` file that would give the pinned `vizro==0.1.34`. But this is undoubtedly a more complex workflow and a best practice ignored by many people. 

Overall I think both the `import vizro` route and the URL one are definitely worth explaining, it's just a question of which one is best for most users for the default recommendation.

AnnMarieW commented:
Yes, you are describing best practices and I think many people have been lax about that - especially since it's so rare for dash to have breaking changes. 

 A lot if people will have issues with Dash 3 because so many community libraries (dbc, dmc, dag, and others) need updating before they will work with dash 3.  I'm currently testing with pre-releases for dash, dmc and dag, which I can't do if I import vizro.

antonymilne commented:
Ah yes, that makes sense - for the purposes of testing these pre-releases, it's definitely easier to just minimise dependencies on other libraries and use the static stylesheet for vizro.

FWIW, when we last checked it seemed like Dash 3 wouldn't actually break anything on Vizro so we should be able to unpin `dash<3` pretty quickly after it's released.

I don't know whether you're a `uv` user, but FYI it makes it much easier to install packages that have incompatible requirements. This can be very useful for testing out pre-releases e.g. I can already do `uv pip install vizro dash==3.0.0rc1` without it complaining at all, and there's a special `--override` flag to force-install packages with incompatible requirements.

AnnMarieW commented:
> FWIW, when we last checked it seemed like Dash 3 wouldn't actually break anything on Vizro

That's surprising, because if you run dash 3 with any dash-ag-grid  < 31.3.1rc1, you should see this error:

> Cannot read properties of undefined (reading 'is_loading')

Also, certain components in dbc don't render correctly such as `Tabs`.  The `Accordion` with components in the titles don't render, plus if you use a `dbc.Spinner` like a `dcc.Loading` component, it doesn't work as expected.  (There are no error message with any of these)  This may affect people who use custom components more than those who stick with pure Vizro apps.

dmc shouldn't have any issues with the date picker component, so it will likely affect only those who use custom components.

Switching gears... I'd like to promote the new Vizro Bootstrap theme and templates on the forum.  I was  thinking about just posting what I have in the new Vizro Bootstrap section from the Theme Explorer site, but if you would like to jazz that up more, I'd be happy to have you lead and I can chime in on your post.

huong-li-nguyen commented:
Hey @AnnMarieW - feel free to do so!! 🙏 

I think @antonymilne was happy for you to take the lead on announcing it first, and we'll probably post another general vizro update after! 👍
